rosidl_generator_java/resource/msg.cpp.em

```diff
@@ -220,72 +209,95 @@ JNIEXPORT jlong JNICALL Java_@(underscore_separated_jni_type_name)_getDestructor
   }
 @[for member in message.structure.members]@
 @{
-normalized_type = get_normalized_type(member.type)
+if isinstance(member.type, AbstractNestedType):
+    normalized_type = get_normalized_type(member.type.value_type)
+    get_java_name = get_java_type(member.type.value_type, use_primitives=True)
+    get_method_name = get_java_type(member.type.value_type, use_primitives=True).capitalize()
+    jni_signature = get_jni_signature(member.type.value_type)
```

Comment:
Good question, I had to double-check myself. We want the primitive name (if the type has one) for declaring array objects and calling related functions, e.g. j**int**Array and New**Int**Array (`get_java_name` and `get_method_name` respectively).

The function `get_jni_signature` ends up mapping from boxed types to JNI primitive signatures [here](https://github.com/ros2-java/ros2_java/blob/0dac87fd1af7a45fda22d276214941fad89f0ea2/rosidl_generator_java/rosidl_generator_java/__init__.py#L189-L196). TBH, I'm not sure if that is the correct mapping, but it is working out in this case. IMO, the mapping should at least account for primitives, but this I think we can make that refactor in a follow-up PR.

Lastly, it looks like `normalized_type` isn't actually used in the template in conditional blocks for primitive (basic) types, so it will be consistent with the other names in cases with object types.